### PR TITLE
Allow a referrer to add current offences

### PR DESCRIPTION
**Pull request by patrickjfl**
# Context

[Trello ticket](https://trello.com/c/GqN6gnPJ/417-current-offence-info-add-a-current-offence)

# Changes in this PR

This PR allows referrers to add current offences to the application.

## Screenshots of UI changes

Screenshot 2023-11-27 at 15 12 47

Screenshot 2023-11-27 at 15 13 01

Screenshot 2023-11-27 at 15 13 43

Screenshot 2023-11-27 at 15 14 06

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/form-pages/apply/offence-and-licence-information/offending-history/custom-forms/offenceHistoryData.ts [hunk 1]
```diff
@@ -44,33 +44,29 @@ export default class OffenceHistoryData implements TaskListPage {
...
 
-  offenceCategories = this.getCategoriesAsItemsForSelect()
+  offenceCategories: Array<SelectItem>
 
   constructor(
     body: Partial<OffenceHistoryDataBody>,
     private readonly application: Cas2Application,
   ) {
     this.body = body as OffenceHistoryDataBody
+    this.offenceCategories = this.getCategoriesAsItemsForSelect(this.body.offenceCategory)
```
Comment (Harriethw) [inline server/form-pages/apply/offence-and-licence-information/offending-history/custom-forms/offenceHistoryData.ts:54]: nice fix

**Review by Harriethw — APPROVED** [review]
niice. Another Task down 😎

File: server/views/applications/pages/current-offences/current-offence-data.njk [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          }
+        },
+        fetchContext()
+      )
+    }}
+{% endset %}
+
+{% block content %}
+
+  <div class="govuk-grid-row">
```
Comment (Harriethw) [inline server/views/applications/pages/current-offences/current-offence-data.njk:26]: perhaps one that could be picked up as a snag, and our other 'data' pages probably don't have this either, but noticed this page doesn't have a `Back` or `back to Task list` link/button.

Comment (patrickjfl) [reply server/views/applications/pages/current-offences/current-offence-data.njk:26]: yes, good idea. Looks like the other ones don't have them either - I'll add a ticket to the snag list

**Pull request merged**